--- Net-Server-2.008/t/NetServerTest.pm	2014-05-13 00:31:32.000000000 +0900
+++ Net-Server-2.008-patched/t/NetServerTest.pm	2016-12-24 10:18:24.850000002 +0900
@@ -67,7 +67,7 @@
     print NST_WRITE "22";
     is(read(NST_READ, my $buf, 2), 2, "Pipe works") || do { SKIP: { skip ("Couldn't use working pipe", $N - 3) }; exit };
     warn "# Checked pipe serialization\n" if debug;
-    $env{'block_until_ready_to_test'} = sub { read(NST_READ, my $buf, 1) };
+    $env{'block_until_ready_to_test'} = sub { sleep 1 if $^O eq 'MSWin32'; read(NST_READ, my $buf, 1) };
     $env{'signal_ready_to_test'}      = sub { print NST_WRITE "1"; NST_WRITE->flush; };
 
     return \%env;
